services/movie_session.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    movie_session = MovieSession.objects.create(
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id,
+        show_time=movie_show_time
+    )
+    return movie_session
+
+
+def get_movies_sessions(session_date: str | None = None) -> QuerySet:
+    new_set = MovieSession.objects.all()
+    if session_date:
+        new_set = new_set.filter(show_time__date=session_date)
+
+    return new_set
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> MovieSession:
+    movie_session = MovieSession.objects.get(id=session_id)
```

Comment:
Can you refactor this function so that you will not need to get movie session if none of the update parameters has been passed?